packages/blocks/src/api/factory.js

```diff
@@ -407,17 +407,6 @@ export function switchToBlockType( blocks, name ) {
 	const firstBlock = blocksArray[ 0 ];
 	const sourceName = firstBlock.name;
 
-	// Unless it's a Grouping Block then for multi block selections
-	// check that all Blocks are of the same type otherwise
-	// we can't run a conversion
-	if (
-		! isContainerGroupBlock( name ) &&
-		isMultiBlock &&
-		! isBlockSelectionOfSameType( blocksArray )
```

Comment:
Pretty sure this function `isBlockSelectionOfSameType` becomes unused after this change (exported, but only for purpose of unit testing).